### PR TITLE
wrapper for api calls

### PR DESCRIPTION
**Pull request by RafeedIqbal**
image
the wrappers seem to be working fine

made the button on the landing page route to the test page since i couldn't figure out how to get to a page manually.

### CODE REVIEW

File: src/poc/server/poc-server/src/controllers/users.ts [hunk 1]
```diff
@@ -86,4 +86,5 @@ export const emitConnection = async (req: Request, res: Response) => {
     const { uuid } = req.body;
     io.to(uuid).emit("event");
     res.status(200).json({ message: "Connection established" });
+    console.log("Connection established for", req.body)
```
Comment (Spitgranger) [inline src/poc/server/poc-server/src/controllers/users.ts:89]: these changes needed?

File: src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+const baseUrl="http://localhost:5001";
+  
+async function apiFetch<T>(endpoint: string, options: RequestInit = {}): Promise<T> {
+    const url = `${baseUrl}/${endpoint}`;
+  
+    const response = await fetch(url, {
```
Comment (Spitgranger) [inline src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx:6]: this should be wrapped in try catch block

Comment (Spitgranger) [reply src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx:6]: lines 6 to 19

File: src/poc/server/poc-server/src/controllers/users.ts (hunk 1 above)
Comment (RafeedIqbal) [reply src/poc/server/poc-server/src/controllers/users.ts:89]: This is just to see if the api call worked, the other two have console messages.

File: src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx (hunk 2 above)
Comment (RafeedIqbal) [reply src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx]: Most guides i saw say that the error is better caught when calling the async function (the throw error part is mainly to reject the promise). I changed the test page to add error handling.

**Review by Spitgranger — APPROVED** [review]
lgtm

**Pull request merged**